odoo/models.py

```diff
@@ -1633,7 +1633,7 @@ def search_fetch(self, domain, field_names, offset=0, limit=None, order=None):
         # first determine a query that satisfies the domain and access rules
         query = self._search(domain, offset=offset, limit=limit, order=order or self._order)
 
-        if query.is_empty():
+        if isinstance(query, list) or query.is_empty():
```

Comment:
The `_search` method should always return a Query, probably it needs to be fixed there instead of here.